Update package to 'dylan-tool'

fraya opened the pull request:
Update package to 'dylan-tool' and fix missing package infrastructure (README, tests, etc).

fraya committed:
Add .gitignore

fraya committed:
Add license

fraya committed:
Add package description

fraya committed:
Add README

fraya committed:
Add tests for 'aif'

fraya committed:
Remove Makefile

fraya committed:
Add CI

fraya committed:
Add build-and-test badge to README

cgay commented on LICENSE.txt:
```diff
@@ -0,0 +1,165 @@
+                   GNU LESSER GENERAL PUBLIC LICENSE
+                       Version 3, 29 June 2007
```
I'm a little wary of changing the licensing of old code. I see that Doug Auclair put "opyright: (c) 2002, LGPL" in the original commit so I wouldn't change it to the 2007 version of LGPL, which could be different.  It looks like LGPL-2.1 was in effect in 2002. Update to that instead?

fraya replied:
My license-foo is weak, I thought that this LGPL was the same. I'll change it.

cgay commented on README.md:
````diff
@@ -0,0 +1,40 @@
+end;
+```
+
+### References
````
This should probably be a second level header. `##`

fraya replied:
Ok.

cgay commented on tests/anaphora-test.dylan:
```diff
@@ -0,0 +1,60 @@
+define suite anaphora-test-suite ()
+  suite aif-test-suite;
+end suite;
+
```
You don't need to define any suites. These tests will run so quickly I can't imagine ever wanting to run only a subset of them.  But it's up to you.

fraya replied:
You are right. I love the ceremony.

cgay commented on tests/library.dylan:
```diff
@@ -0,0 +1,21 @@
+define module anaphora-test
+  use common-dylan;
+  use common-extensions,
+    import: { integer-to-string };
```
Everything in `common-extensions` is reexported from `common-dylan` so you shouldn't need this separate import.

fraya replied:
Thank you. I'll change it.

cgay reviewed (commented):
Looks good with a few minor changes. Nice to have tests and some doc!

fraya committed:
Change LGPL for v2.1

Use the same LGPL version that was in use in 2002.

fraya committed:
Change section level in 'References'

fraya committed:
Remove import not needed for module 'common-extensions'

Everything in 'common-extensions' is reexported from 'common-dylan' so
I don't need to import the module 'common-dylan' and 'common-extensions'.
Thanks to @cgay
